lib/systems/lobby_system.cpp

```diff
@@ -0,0 +1,148 @@
+#include "systems/lobby_system.hpp"
+
+#include <algorithm>
+
+#include "brickengine/input.hpp"
+#include "player_input.hpp"
+#include "components/character_selection_component.hpp"
+#include "brickengine/components/renderables/texture_component.hpp"
+#include "brickengine/components/transform_component.hpp"
+#include "brickengine/components/player_component.hpp"
+
+LobbySystem::LobbySystem(std::shared_ptr<EntityFactory> ef, std::shared_ptr<EntityManager> em) : BeastSystem(ef, em), ef(ef), em(em) {};
+
+void LobbySystem::update(double) {
+    auto& input = BrickInput<PlayerInput>::getInstance();
+
+    auto selector_entities = em->getEntitiesByComponent<CharacterSelectionComponent>();
+
+    for(auto& [entity_id, selector] : selector_entities) {
+        if(input.checkInput(selector->player_id, PlayerInput::GRAB)) {
+            if(!selector->joined) {
+                // Player wants to play
+                selector->joined = true;
+                for(auto& character : picked_characters) {
+                    // Select the first available character
+                    if(!character.second) {
+                        selector->selected_character = character.first;
+                        ef->changeCharacterSelectorTexture(entity_id, character.first, true);
+                        break;
+                    }
+                }
+            } else {
+                // Player selected character
+                if(!selector->picked) {
+                    // Check if character is available
+                    bool available = true;
+                    for(auto character : picked_characters) {
+                        if(character.first == selector->selected_character && character.second) {
+                            available = false;
+                            break;
+                        }
+                    }
+
+                    if(available) {
+                        if(selector->selected_character != Character::RANDOM) {
+                            for(auto& character : picked_characters) {
+                                if(character.first == selector->selected_character) {
+                                    character.second = true;
+                                    break;
+                                }
+                            }
+
+                            // Update the other selector so they do not show the character chosen by this player
+                            auto selector_entities_2 = em->getEntitiesByComponent<CharacterSelectionComponent>();
```

Comment:
I see something closely related to collision_detector2 👀. All jokes aside, I would call this something like; 'to_update_select_entities' to give a bit more context what's about to happen to (some) of the character select components.

Reply:
I could not think of a better name yesterday but "to_update" will be more clear for sure.

The funny thing is, this is not so closely related to "collision_detector2" because 1 exists in this code :wink: